web/src/routes/admin/library-management/+page.svelte

```diff
@@ -393,12 +406,19 @@
                     <Icon path={mdiDotsVertical} size="16" />
                   </button>
 
-                  {#if showContextMenu}
+                  {#if showContextMenu && selectedLibrary}
                     <Portal target="body">
                       <ContextMenu {...contextMenuPosition} on:outclick={() => onMenuExit()}>
                         <MenuOption on:click={() => onRenameClicked()} text={`Rename`} />
-
-                        {#if selectedLibrary && selectedLibrary.type === LibraryType.External}
+                        <MenuOption
+                          on:click={() => onToggleReadOnlyClicked()}
+                          text={selectedLibrary.isReadOnly ||
+                          (selectedLibrary.isReadOnly == null && selectedLibrary.type === LibraryType.External)
```

Comment:
When would `isReadOnly` be null-ish?

Reply:
yeah, it shouldn't be (assuming we do the migration correctly). I've updated it to remove that